mirrord/config/src/feature/database_branches.rs

```diff
@@ -0,0 +1,143 @@
+use std::ops::Deref;
+
+use mirrord_analytics::{Analytics, CollectAnalytics};
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use crate::config::{ConfigContext, FromMirrordConfig, MirrordConfig};
+
+/// A list of configuration for database branches.
+///
+/// ```json
+/// {
+///   "feature": {
+///     "db_branches": [
+///       {
+///         "name": "my-database-name",
+///         "ttl_secs": 120,
+///         "type": "mysql",
+///         "version": "8.0",
+///         "connection": {
+///           "url": {
+///             "type": "env",
+///             "variable": "DB_CONNECTION_URL"
+///           }
+///         }
+///       }
+///     ]
+///   }
+/// }
+/// ```
+#[derive(Clone, Debug, Eq, PartialEq, JsonSchema, Serialize, Deserialize, Default)]
+pub struct DatabaseBranchesConfig(pub Vec<DatabaseBranchConfig>);
+
+impl Deref for DatabaseBranchesConfig {
+    type Target = Vec<DatabaseBranchConfig>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl DatabaseBranchesConfig {
+    pub fn mysql(&self) -> impl '_ + Iterator<Item = &DatabaseBranchConfig> {
+        self.0.iter().filter(|db| db._type == DatabaseType::MySql)
+    }
+}
+
+/// Configuration for a database branch.
+///
+/// By specifying a unique `id`, the branch database will be reused if not expired.
+///
+/// Example:
+///
+/// ```json
+/// {
+///   "id": "my-branch-db",
+///   "name": "my-database-name",
+///   "ttl_secs": 120,
+///   "type": "mysql",
+///   "version": "8.0",
+///   "connection": {
+///     "url": {
+///       "type": "env",
+///       "variable": "DB_CONNECTION_URL"
+///     }
+///   }
+/// }
+/// ```
+#[derive(Clone, Debug, Eq, PartialEq, JsonSchema, Serialize, Deserialize)]
+pub struct DatabaseBranchConfig {
+    /// A unique identifier chosen by the user. This is useful when reusing branch database
+    /// sharing among different Kubernetes users.
+    pub id: Option<String>,
```

Comment:
In the issue this is called `db_branch_id`. @0x00A5, id you discuss a change with @liron-sel?

Reply:
Updated the ticket